instrumentation/jsf/jsf-mojarra-3.0/javaagent/build.gradle.kts

```
@@ -0,0 +1,30 @@
+plugins {
+  id("otel.javaagent-instrumentation")
+  id("org.unbroken-dome.test-sets")
+}
+
+muzzle {
+  pass {
+    group.set("org.glassfish")
+    module.set("jakarta.faces")
+    versions.set("[3,)")
+    extraDependency("jakarta.el:jakarta.el-api:4.0.0")
+    assertInverse.set(true)
+  }
+}
+
+otelJava {
+  minJavaVersionSupported.set(JavaVersion.VERSION_11)
+}
+
+dependencies {
+  compileOnly("jakarta.faces:jakarta.faces-api:3.0.0")
+  compileOnly("jakarta.el:jakarta.el-api:4.0.0")
+
+  implementation(project(":instrumentation:jsf:jsf-jakarta-common:javaagent"))
+
+  testImplementation("org.glassfish:jakarta.faces:3.0.4")
```

```suggestion
  testLibrary"org.glassfish:jakarta.faces:3.0.4")
```

We have a separate `testLatestDeps` test task that runs the tests after bumping the `library` and `testLibrary` dependencies to their latest versions. You could use that and cover both mojarra 3 & 4

---

Unfortunately, the test will not run with JSF 4 implementation as it requires CDI Injector instead of BeanManager, I have tried to add Weld CDI implementation but wasn't able to make it work with embedded Jetty (I have no experience with it, maybe someone can add that later).

But from other aspects, JSF 3 and JSF 4 are mostly the same, and it works without any issues in my project with JSF 4 (MyFaces).

---

Ah, I see. Can you add a 
```kotlin
latestDepTestLibrary("org.glassfish:jakarta.faces:3.+")
```
line that limits the version to 3.x; and a comment that explains the reason?